samples/subsys/logging/logger/src/main.c

```diff
@@ -189,15 +194,18 @@ static void external_log_system_showcase(void)
 	ext_log_system_foo();
 }
 
-void main(void)
+static void wait_on_log_flushed(void)
 {
-	int err = LOG_INIT();
-
-	if (err == 0) {
-		err = log_set_timestamp_func(timestamp_get, timestamp_freq());
+	while (log_buffered_cnt()) {
+		k_sleep(5);
 	}
+}
 
-	(void)err;
+void log_demo_thread(void *dummy1, void *dummy2, void *dummy3)
+{
+	k_sleep(100);
```

Comment:
Why this k_sleep() is needed?

Reply:
it can be removed once #8842 goes in. It's about allowing log to initialize since log thread is on low priority and initialization is there.

Reply:
@nashif: Could you please merge #8842?
@nordic-krch: Could you please rebase this after #8842 will be merged?

Reply:
done